docs/architecture/state.md

```diff
@@ -1,91 +1,111 @@
-Miden rollup state describes the current condition of all accounts and notes; i.e. what is currently the case.
+# State
 
-As the state model uses concurrent off-chain state, Polygon Miden aims for private, and parallel transaction execution and state bloat minimization.
+## What is the purpose of state?
 
-Miden's goals include:
+Miden enables secure, private, and scalable transaction execution. By employing a concurrent state model with local execution and proving, Miden achieves three primary objectives: preserving privacy, supporting parallel transactions, and minimizing on-chain data storage (state bloat).
 
-* Notes and nullifiers to ensure privacy of note consumption.
-* Flexible data storage for users who can store their data off-chain or with the network.
-* Parallel transactions executed concurrently by distinct actors.
-* Concurrent state model that allows block production without knowing the full state.
+Miden’s state model focuses on:
 
-Privacy is enforced by a UTXO-like state model consisting of notes and nullifiers combined with off-chain execution using zero-knowledge proofs.
+- **Privacy:**  
+  By using notes and nullifiers, Miden ensures that value transfers remain confidential. Zero-knowledge proofs allow users to prove correctness without revealing sensitive information.
 
-State bloat describes the ever growing state stored in blockchain nodes. Polygon Miden addresses this challenge via its state model that enables concurrent off-chain execution and off-chain storage. Simply put, users can store their own data locally which reduces the burden on the network while integrity is ensured using zero-knowledge.
+- **Flexible data storage:**  
+  Users can store data privately on their own devices or within the network. This approach reduces reliance on the network for data availability, helps maintain user sovereignty, and minimizes unnecessary on-chain storage.
 
-## State components
+- **Concurrency:**  
+  Multiple transactions can be processed concurrently by distinct actors which improves throughput and efficiency.
 
-Miden nodes maintain three databases to describe state:
+Miden’s state model supports a private, secure, and high-throughput environment while also addressing the challenges of state bloat commonly associated with continuously growing blockchain states.
 
-1. A database of accounts.
-2. A database of notes.
-3. A database of nullifiers for already consumed notes.
+## What is state?
 
-![Architecture core concepts](../img/architecture/state/state.png)
+The state of the Miden rollup describes the current condition of all accounts and notes in the protocol; i.e., the current reality.
```

Comment:
Probably not for this PR, but I think we should come up with a better wording then just "state". Maybe we should say "ledger state", "blockchain state" or something like that.